### ISSUE TITLE
Generic metaclass should be flagged as error

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#2683]
```python
from typing import TypeVar, Generic

T = TypeVar("T")

class FooMetaclass(type, Generic[T]): pass

class Foo(metaclass=FooMetaclass[T]): # This should result in an error
    pass
```

This came up in the following mypy issue: https://github.com/python/mypy/issues/11672#issuecomment-989441935

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.194, which I just published. It will also be included in the next release of pylance.
